### PR TITLE
Creating of Product/Robot posts

### PR DESCRIPTION
**Pull request by glinxtech**
Please fix my javascript, oh great javascript overlord Thomas. (Don't be too harsh tho, thx)

### CODE REVIEW

File: WebdevPeriod3/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -78,12 +73,12 @@ public async Task<IActionResult> Profile()
         {
             
             // RETRIEVE USER INFORMATION
-            User user = await _userManager.GetUserAsync(User);
+            var user = await _userManager.GetUserAsync(User);
```
Comment (stefvanhouten) [inline WebdevPeriod3/Controllers/DashboardController.cs]: Why did you change this with var?

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+}
+
+let ctrl = null;
+
+async function search(term) {
+  if (ctrl) ctrl.abort();
+  ctrl = new AbortController();
+
+  const products = await fetch(`../products?searchterm=${term}`, {
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: Here you are using a mixture of async and await and promises. Please choose one or the other and do not combine both.

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    signal: ctrl.signal
+  })
+    .then((res) => res.json())
+    .catch((ex) => (ex.name === "AbortError" ? null : Promise.reject(ex)))
+    .finally(() => {
+      ctrl = null;
+    });
+
+    console.log(products);
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: Debugging leftover?

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    product.name.toLowerCase()
+  );
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+
+  let terms;
+
+    document.getElementById("search-input").addEventListener("input", debounce(async (e) => {
+    if (e.target.value)
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: Naked if statements are bad practice, if someone wants to adjust your code later on they have to add brackets everywhere. Especially here you should use brackets, you might get away with the other occurences.

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		  terms = await search(e.target.value);
+		if (terms) {
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: Before you used naked if statements, if you do so please be consistent.

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+			document.getElementById("search-output").innerHTML = terms.slice(0,5).filter((term) => term.name != e.target.value).map((term) => `<li>${term.name}</li>`).join("");
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: The usage of innerHTML can bring a potential security risk(XSS) when using this with user input. I'm not sure if thats the case here

File: WebdevPeriod3/wwwroot/js/subsystemlist.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    element.addEventListener('click', () => {
+      document.getElementById("search-input").value = this.value;
+    });
+  });
+
+  document.getElementById("search-output").addEventListener("click", (e) => {
+    document.getElementById("search-input").value = e.target.innerHTML;
+  })
+
+  document.getElementById("add-system").addEventListener("click", () => {
```
Comment (stefvanhouten) [inline WebdevPeriod3/wwwroot/js/subsystemlist.js]: This is a pretty big function, might want to split it up and make sure each function does one thing.

File: WebdevPeriod3/wwwroot/js/subsystemlist.js (hunk 6 above)
Comment (glinxtech) [reply WebdevPeriod3/wwwroot/js/subsystemlist.js]: Had read the same thing, but seems to be harmless here.

File: WebdevPeriod3/Controllers/DashboardController.cs (hunk 1 above)
Comment (glinxtech) [reply WebdevPeriod3/Controllers/DashboardController.cs]: I wrote it to begin with. No reason to type out User anyways when var is available

**Comment by thomasio101** [comment]
I'll give this a touchup over the coming days, okay?

**Comment by thomasio101** [comment]
@glinxtech, waarom toont de zoekfunctie niet een lijst met zoekresultaten?

Het is niet gebruikersvriendelijk om te verwachten dat de gebruikers de namen van hun producten uit hun hoofd kennen.

**Review by thomasio101 — APPROVED** [review]
Looks good, though there are still some improvements to be made to the HTML, especially regarding accessibility.

We also need to add authorization to some of the new routes.

I'll create issues to address these concerns later.
